server/src/com/cloud/dc/dao/DedicatedResourceDaoImpl.java

```diff
@@ -312,6 +317,65 @@ public DedicatedResourceVO findByHostId(Long hostId) {
     }
 
     @Override
+    public List<DedicatedResourceVO> listAvailableResources(Long accountId, Long... domains) {
```

Comment:
Please rewrite this using the standard query builder (SearchBuilder) that is available and used in the other methods in this class.